packages/flutter_tools/test/integration.shard/build_ios_config_only_test.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:process/process.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+import '../src/common.dart';
+
+void main() {
+  test('flutter build ios --config only updates generated xcconfig file without performing build', () async {
+    final String woringDirectory = globals.fs.path.join(getFlutterRoot(), 'examples', 'hello_world');
+    final String flutterBin = globals.fs.path.join(getFlutterRoot(), 'bin', 'flutter');
+
+    await const LocalProcessManager().run(<String>[
+      flutterBin,
+      'clean',
+    ], workingDirectory: woringDirectory);
+    final ProcessResult result = await const LocalProcessManager().run(<String>[
+      flutterBin,
+      'build',
+      'ios',
+      '--config-only',
+      '--release',
+      '--obfuscate',
+      '--split-debug-info=info',
```

Comment:
can you also add an "advanced" usage that synergizes with xcode more? Something like --flavor.

Reply:
`--flavor` does not change any of the generated config, it just controls which scheme we tell xcodebuild to use.

Reply:
true dat. How about --local-engine?

Reply:
we'll validate that it doesn't exist and fail :)

Reply:
as in you're going to assert users don't flutter build ios --config-only with --local-engine? Should we? If this is a flutter build ios --config-only command rather a flutter build ipa command, the intent is broader now no? We can't tell what the user wants to do and flutter build ios --config-only --local-engine can be a valid intent (e.g. this new path is now probably going to become the easiest way to flutter build ios --config-only --local-engine, then launch xcode and put a breakpoint on the engine code).

Reply:
no, the tool will fail to locate the local engine because it does not exist.

Reply:
ah got it. I misunderstood.